src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -52,24 +61,30 @@ internal class ExpressionVariableFinder : CSharpSyntaxWalker
 
             var finder = s_poolInstance.Allocate();
             finder._scopeBinder = binder;
+            finder._enclosingBinder = binder;
             finder._localsBuilder = builder;
 
             foreach (var n in nodes)
             {
-                finder.Visit(n);
+                finder._nodeToBind = n;
+                finder.VisitNodeToBind(n);
             }
 
+            finder._nodeToBind = null;
```

Comment:
I don't think it makes a difference, as it is reset to its previous value by VisitNodeToBind. I'll remove this.

---
In reply to: [76321908](https://github.com/dotnet/roslyn/pull/13274#discussion_r76321908) [](ancestors = 76321908)